gradio/processing_utils.py

```diff
@@ -289,9 +289,11 @@ def _move_to_cache(d: dict):
     return client_utils.traverse(data, _move_to_cache, client_utils.is_file_obj)
 
 
-def add_root_url(data, root_url) -> dict:
+def add_root_url(data: dict, root_url: str, previous_root_url: str | None) -> dict:
     def _add_root_url(file_dict: dict):
         if not client_utils.is_http_url_like(file_dict["url"]):
+            if previous_root_url and file_dict["url"].startswith(previous_root_url):
```

Comment:
Why do we need this extra step here? If we're running this we know that the root changed so we can just replace the root key entirely. I think we can get rid of the previous_root_url altogether.

Reply:
Yes but this allows us to robustly identify what part of the `file_dict["url"]` is the previous root and strip it away. Otherwise, we'd have to look for the presence of a prefix before `/file=`, which makes me a bit nervous in case we change that prefix or somehow we have an unusual path

Reply:
Gotcha makes sense!